h2o-k8s/src/main/java/water/k8s/probe/KubernetesLeaderNodeProbeHandler.java

```diff
@@ -0,0 +1,41 @@
+package water.k8s.probe;
+
+import fi.iki.elonen.NanoHTTPD;
+import fi.iki.elonen.router.RouterNanoHTTPD;
+import water.H2O;
+import water.k8s.KubernetesEmbeddedConfigProvider;
+import water.util.Log;
+
+import java.util.Map;
+
+
+public class KubernetesLeaderNodeProbeHandler extends RouterNanoHTTPD.DefaultHandler {
+
+    @Override
+    public String getText() {
+        throw new IllegalStateException(String.format("Method getText should not be called on '%s'",
+                getClass().getName()));
+    }
+
+    @Override
+    public String getMimeType() {
+        return "text/plain";
+    }
+
+    @Override
+    public NanoHTTPD.Response.IStatus getStatus() {
+        throw new IllegalStateException(String.format("Method getMimeType should not be called on '%s'",
+                getClass().getName()));
+    }
+
+    @Override
+    public NanoHTTPD.Response get(RouterNanoHTTPD.UriResource uriResource, Map<String, String> urlParams, NanoHTTPD.IHTTPSession session) {
+        // All nodes report ready state until the clustering process is finished. Since then, only the leader node is ready.
+        if (H2O.SELF == null || H2O.SELF.isLeaderNode() || !KubernetesEmbeddedConfigProvider.isClustered()) {
```

Comment:
H2O.SELF should theoretically never transition from a valid reference to null. However, to handle edge cases better I like to introduce a local variable to avoid possible NPEs:

H2ONode self = H2O.SELF;

self == null | self.isLeaderNode || ...

Reply:
Great, thanks. Fixed in https://github.com/h2oai/h2o-3/pull/4554/commits/22be4dc4fff2fbe83e14f4dc9b4de2718e30e275.